doc/source/data_structures.rst

```diff
@@ -1,82 +1,516 @@
+
+
+Topology-specific methods
+-------------------------
+
+A number of analysis and transformation methods are defined for :code:`AtomGroup` that require specific properties to be available. The primary requirement is usually coordinates, or the `positions` attribute. With coordinates, you can easily compute a center of geometry:
+
+.. code-block::
+
+    >>> u.atoms.center_of_geometry()
+    array([-0.04223882,  0.01418196, -0.03504874])
+
+When atom masses are defined, a number of other methods become available. These include:
+
+* :func:`center_of_mass`
+* :func:`radius_of_gyration`
+* :func:`moment_of_inertia`
+* :func:`asphericity`
+* :func:`align_principal_axes`
+
+See `the Topology system <../topology_system.html>`_ for more information about which attributes are required for which :code:`AtomGroup` methods.
+
+
+Groups of atoms
+===============
+
+MDAnalysis has a hierarchy of :code:`Atom` containers that are used throughout the code.
+
+.. image:: images/classes.png
+
+First and foremost is the :code:`AtomGroup`. An :code:`AtomGroup` is the primary :code:`Atom` container; virtually everything can be accessed through it, as detailed `above <https://www.mdanalysis.org/UserGuide/data_structures.html#atomgroup>`_. This includes chemically meaningful groups of :code:`Atom`\ s such as a :code:`Residue` or a :code:`Segment`. 
+
+---------------------
+Residues and Segments
+---------------------
+
+A :code:`Residue` is composed of :code:`Atom`\ s, and a :code:`Segment` is composed of :code:`Residue`\ s.
+
+The corresponding container groups are :code:`ResidueGroup` and :code:`SegmentGroup`. These have similar properties and available methods as :code:`AtomGroup`.
+
+Each of these container groups can be accessed through another. For example:
+
+.. code-block::
+
+    >>> ag = u.atoms.select_atoms('resname ARG and name CA')
+    >>> ag
+    <AtomGroup with 13 atoms>
+    >>> ag.residues
+    <ResidueGroup with 13 residues>
+    >>> ag.residues.atoms
+    <AtomGroup with 312 atoms>
+    >>> ag.segments
+    <SegmentGroup with 1 segment>
+    >>> ag.segments.atoms
+    <AtomGroup with 3341 atoms>
+
+Similarly, an :code:`Atom` has direct knowledge of the :code:`Residue` and :code:`Segment` it belongs to.
+
+.. code-block::
+
+    >>> a = u.atoms[0]
+    >>> a.residue
+    <Residue MET, 1>
+    >>> a.residue.segment
+    <Segment 4AKE>
+    >>> a.residue.segment.residues
+    <ResidueGroup with 214 residues>
+
+-----------------------
+Fragments and molecules
+-----------------------
 
+Unlike Residues and Segments, fragments and molecules are not classes in MDAnalysis. A fragment is defined by bond connectivity. A fragment is what is typically considered a molecule: a group of atoms where each atom is bonded to at least one other atom in the fragment, and are not bonded to any atoms outside the fragment. The fragments of a Universe are determined by MDAnalysis as a derived quantity.
```

Comment:
Not sure what you mean by "are not classes in MDA"? This sounds as if the user needs to understand an implementation detail. Or do you want to make clear that fragments are somewhat different from the standard hierarchy? I think the clearest way to show the difference is by way of examples, as you're doing below.

Reply:
I took that out. I like the discussion of more detailed developer docs in MDAnalysis/mdanalysis#1175 . Keep the picture Y/N?

![classes](https://user-images.githubusercontent.com/31115101/67347074-c15f8500-f58c-11e9-8f44-e36222976ed6.png)

Reply:
Keep (with minor modification) – these are basic containers and they have a "is contained in" hierarchical and exclusive relationship, which is really nicely illustrated. But remove "molecule" as a (currently) Gromacs/TPR specific concept. If fragments are not discussed here, say that these are an advanced concept and discussed elsewhere (link).